flow/templates/delta.sh

```diff
@@ -0,0 +1,65 @@
+{# Templated in accordance with: https://https://wiki.ncsa.illinois.edu/display/DSC/Delta+User+Guide #}
+{% extends "slurm.sh" %}
+{% block tasks %}
+    {% set threshold = 0 if force else 0.9 %}
+    {% set cpu_tasks = operations|calc_tasks('np', parallel, force) %}
+    {% set gpu_tasks = operations|calc_tasks('ngpu', parallel, force) %}
+    {% if gpu_tasks %}
+        {% if partition == "gpuA40x4" %}
+            {% set nn_gpu = gpu_tasks|calc_num_nodes(4) %}
+            {% if nn_gpu > 100 %}
+                {% raise "More nodes requested than exist on the system." %}
+            {% endif %}
+        {% elif partition == "gpuA100x4" %}
+            {% set nn_gpu = gpu_tasks|calc_num_nodes(4) %}
+            {% if nn_gpu > 100 %}
+                {% raise "More nodes requested than exist on the system." %}
+            {% endif %}
+        {# We do not allow submission to the partitions below as they have few #}
```

Comment:
I think typically we support all partitions. I was told explicitly not to in this case by @joaander, so I wanted any users to not be given an error message that this is intentional when it happens.

Reply:
We do typically list all partitions. I don't have a strong opinion here, although I will say that if we don't list partitions users might make a feature request to add them whereas with this solution they get a clear error on why.

On that note, it might help to make these errors contain a bit more explanation of why we don't support these partitions to head off such questions ;)

Reply:
Check the docs. The x8 partitions have a *very* limited number of nodes and charge extra SUs to use. These *highly* specialized nodes are only useful for specific machine learning models that are only able to scale to 8 local GPUS but not further over MPI.

To replicate the long discussion we already had on slack 1) I do not want users to accidentally run on the x8 partitions, only to see long queue wait tines coupled with double the necessary SU charge. And 2) I don't expect any signac-flow users are running the highly specialized codes that perform more than 2x faster PER GPU on the x8 nodes vs the x4 nodes. If my expectation is incorrect, then users can always write a custom template.

The MI100 partition has only a **single** node and is intended only for testing and development of AMD GPU codes.

Reply:
The discussion here was not on whether or not to support all partitions, but how we convey a lack of support for specific partitions. The debate was over whether to include the unsupported partitions in the CLI and error at the point of template generation, or if we should simply drop them from the CLI. I was indicating (weak) support for the former since it allows us to give a more informative reason for the error rather than making it seem like an unimplemented feature for which a feature request would be entertained. I trusted that there are good reasons not to support these partitions, but thanks for the context.